rust/operator/src/hdfs_controller.rs

```diff
@@ -1,55 +1,177 @@
-        ServicePort, ServiceSpec, Volume,
+
+use crate::container::ContainerConfig;
+use snafu::{OptionExt, ResultExt, Snafu};
+use stackable_hdfs_crd::{constants::*, HdfsCluster, HdfsPodRef, HdfsRole, MergedConfig};
+use stackable_operator::{
+    builder::{
+        ConfigMapBuilder, ObjectMetaBuilder, PodBuilder, PodSecurityContextBuilder, VolumeBuilder,
     },
+    client::Client,
+    cluster_resources::ClusterResources,
+    commons::product_image_selection::ResolvedProductImage,
+    k8s_openapi::{
+        api::{
+            apps::v1::{StatefulSet, StatefulSetSpec},
+            core::v1::{
+                ConfigMap, ConfigMapVolumeSource, EmptyDirVolumeSource, Service, ServicePort,
+                ServiceSpec,
+            },
+        },
+        apimachinery::pkg::{api::resource::Quantity, apis::meta::v1::LabelSelector},
+    },
+    kube::{
+        api::ObjectMeta,
+        runtime::{controller::Action, reflector::ObjectRef},
+        Resource, ResourceExt,
+    },
+    labels::role_group_selector_labels,
+    logging::controller::ReconcilerError,
+    product_config::{types::PropertyNameKind, ProductConfigManager},
+    product_config_utils::{transform_all_roles_to_config, validate_all_roles_and_groups_config},
+    product_logging::{
+        self,
+        spec::{
+            ConfigMapLogConfig, ContainerLogConfig, ContainerLogConfigChoice,
+            CustomContainerLogConfig,
+        },
+    },
+    role_utils::RoleGroupRef,
 };
-use stackable_operator::k8s_openapi::apimachinery::pkg::apis::meta::v1::LabelSelector;
-use stackable_operator::k8s_openapi::apimachinery::pkg::util::intstr::IntOrString;
-use stackable_operator::kube::api::ObjectMeta;
-use stackable_operator::kube::runtime::controller::Action;
-use stackable_operator::kube::runtime::events::{Event, EventType, Recorder, Reporter};
-use stackable_operator::kube::runtime::reflector::ObjectRef;
-use stackable_operator::kube::{Resource, ResourceExt};
-use stackable_operator::labels::role_group_selector_labels;
-use stackable_operator::memory::to_java_heap;
-use stackable_operator::product_config::{types::PropertyNameKind, ProductConfigManager};
-use stackable_operator::product_config_utils::{
-    transform_all_roles_to_config, validate_all_roles_and_groups_config,
-};
-use stackable_operator::role_utils::RoleGroupRef;
 use std::{
     collections::{BTreeMap, HashMap},
     str::FromStr,
     sync::Arc,
     time::Duration,
 };
+use strum::{EnumDiscriminants, IntoStaticStr};
 
 const RESOURCE_MANAGER_HDFS_CONTROLLER: &str = "hdfs-operator-hdfs-controller";
 const HDFS_CONTROLLER: &str = "hdfs-controller";
 const DOCKER_IMAGE_BASE_NAME: &str = "hadoop";
 
+pub const MAX_LOG_FILES_SIZE_IN_MIB: u32 = 10;
+const OVERFLOW_BUFFER_ON_LOG_VOLUME_IN_MIB: u32 = 1;
+// We have a maximum of 3 continuous logging files and 2 * 2 (stderr, stdout) for init containers (namenode)
+// - name node main container (1)
+// - zkfc side container (1)
+// - vector side container (1)
```

Comment:
The Vector sidecar container does not create a log file. The logs are collected in-memory and forwarded to the aggregator.

Reply:
https://github.com/stackabletech/hdfs-operator/pull/290/commits/06904ae8c965c7bcdf8db08830c5de1262fb6825